java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertToVarHint.java

```diff
@@ -159,8 +191,9 @@ private static boolean preConditionChecker(HintContext ctx) {
         }
 
         // hint is not applicable for compound variable declaration.
-        if (info.getTreeUtilities().isPartOfCompoundVariableDeclaration(treePath.getLeaf()))
+        if (info.getTreeUtilities().isPartOfCompoundVariableDeclaration(treePath.getLeaf())) {
             return false;
+        }
 
         //  hint is not applicable for  variable declaration where type is already 'var'
         return !info.getTreeUtilities().isVarType(treePath);
```

Comment:
for enhanced `for` loop, this helper loops back to the nearest textually preceding assignment, definitely not good ! it seems that the helper need to stop at the boundary of some Tree parent, e.g. statements, variable decl, block, lambda, ...

Reply:
@sdedic , that is an excellent observation. The hint failed to show up in such a scenario. I have fixed this in the latest commit and added tests for it too both in case of var to explicit, and explicit to var hint.